dali/operators/decoder/inflate/inflate_params.h

```diff
@@ -0,0 +1,397 @@
+
+enum class InflateAlg {
+  LZ4
+};
+
+inline std::string to_string(const InflateAlg &alg) {
+  switch (alg) {
+    case InflateAlg::LZ4:
+      return "LZ4";
+    default:
+      return "<unknown>";
+  }
+}
+
+inline InflateAlg parse_inflate_alg(std::string str) {
+  std::transform(str.begin(), str.end(), str.begin(), [](auto c) { return std::tolower(c); });
+  if (str == "lz4") {
+    return InflateAlg::LZ4;
+  }
+  DALI_FAIL(make_string("Unknown inflate algorithm was specified for `", algArgName,
+                        "` argument: `", str, "`."));
+}
+
+template <typename Backend>
+class ShapeParams {
+ public:
+  explicit ShapeParams(const OpSpec &spec)
+      : spec_{spec},
+        shape_{shapeArgName, spec},
+        chunk_offsets_{offsetArgName, spec},
+        chunk_sizes_{sizeArgName, spec},
+        layout_{spec.GetArgument<TensorLayout>(layoutArgName)},
+        sequence_extent_{spec.GetArgument<TensorLayout>(sequenceLayoutArgName)} {
+    DALI_ENFORCE(sequence_extent_.size() == 1,
+                 make_string("The `", sequenceLayoutArgName, "` must be a single character, got `",
+                             sequence_extent_, "`."));
+  }
+
+  void ProcessInputArgs(const Workspace &ws, int batch_size) {
+    SetupOffsetsAndSizes(ws);
+    shape_.Acquire(spec_, ws, batch_size, ArgValue_EnforceUniform, ArgShapeFromSize<1>{});
+    SetupOutputShape(shape_.get());
+    SetupOutputLayout();
+  }
+
+  auto GetMaxOutChunkVol() const {
+    return max_output_sample_vol_;
+  }
+
+  auto GetTotalChunksNum() const {
+    auto total_chunks_num = sizes_.size();
+    assert(total_chunks_num == offsets_.size());
+    assert(total_chunks_num == chunks_per_sample_.num_elements());
+    return total_chunks_num;
+  }
+
+  const auto &GetChunksNumPerSample() const {
+    return chunks_per_sample_;
+  }
+
+  const TensorListShape<> &GetOutputShape() const {
+    return output_shape_;
+  }
+
+  const TensorLayout &GetOutputLayout() const {
+    return output_layout_;
+  }
+
+  const std::vector<int64_t> &GetInChunkOffsets() const {
+    return offsets_;
+  }
+
+  const std::vector<size_t> &GetInChunkSizes() const {
+    return sizes_;
+  }
+
+  bool HasChunks() const {
+    return HasExplicitChunkOffsets() || HasExplicitChunkSizes();
+  }
+
+ private:
+  bool HasExplicitChunkOffsets() const {
+    return chunk_offsets_.HasExplicitValue();
+  }
+
+  bool HasExplicitChunkSizes() const {
+    return chunk_sizes_.HasExplicitValue();
+  }
+
+  void SetupOutputLayout() {
+    if (layout_.empty()) {
+      return;
+    }
+    auto sample_dim = GetOutputShape().sample_dim() - HasChunks();
+    DALI_ENFORCE(
+        layout_.size() == sample_dim,
+        make_string("The size of inflated chunks' layout must match the number of extents in "
+                    "the output shape. However, got the layout `",
+                    layout_, "` while the chunks output shape has ", sample_dim, " extent(s)."));
```

Comment:
This message is quite convoluted. How about just:
```suggestion
        make_string("The layout \"", layout_, "\" has a different number of dimensions (", layout_.size(),
                    ") than the requested output shape (", sample_dim, ")."));
```

Reply:
Sure, thanks.